packages/os/ephemeral-storage.rules

```diff
@@ -0,0 +1,27 @@
+# ephemeral storage links: /dev/disk/ephemeral
+
+ACTION=="remove", GOTO="ephemeral_storage_end"
+SUBSYSTEM!="block", GOTO="ephemeral_storage_end"
+ENV{DEVTYPE}!="disk", GOTO="ephemeral_storage_end"
+
+# Known sources of ephemeral disks:
+# - EC2 instance types with instance storage volumes
+# - KVM VMs with additional virtio disks
+KERNEL!="nvme*|vd*", GOTO="ephemeral_storage_end"
+
+# We can't be sure that devices with similar kernel names are all ephemeral.
+# Our udev helper checks for known system partitions to classify the device.
+IMPORT{program}=="/usr/bin/ghostdog scan $devnode"
+ENV{BOTTLEROCKET_DEVICE_TYPE}!="ephemeral", GOTO="ephemeral_storage_end"
+
+# EBS volumes will show up as "Amazon Elastic Block Store", but we expect these
+# to be either system volumes or managed by the EBS CSI driver.
+KERNEL=="nvme*", ATTRS{model}=="Amazon EC2 NVMe Instance Storage*", ENV{ID_SERIAL}=="?*", \
```

Comment:
Is this only NVME EBS volumes?  What about instance types that do not support NVME devices or EBS volumes that might present differently (like the `st1`, `sc1`, or magnetic types)?

Reply:
EBS volumes of all types won't match this rule, so we won't write the symlink.

Older instance types where the xen-blkfront driver is used for both EBS and ephemeral disks are a problem, since there's no easy way to tell them apart.

We could query IMDS to find the block device mapping and then try to match devices that way. This is complicated because the devices show up before the network is up. We'd have to do the query at a later point during boot, save the mapping somewhere, and trigger a change event on the devices so they get processed again.

I can backlog an issue for this, though I'd prefer to wait until someone says they need it.

Reply:
Opened #1181 so we can track any requests for this feature on older instance types, and to capture my design thoughts.